CDCgov/prime-reportstream
# Support for Bulk Delivery Facilities

This PR adds an endpoint to fetch delivery facilities in bulk, and also updates the item lineage structure to not require recursive logic on-demand. It also includes sorting and date filtering for queries to this endpoint.

Test Steps:
1. Run new migrations, verify the `origin_report_id` and `origin_report_index` columns are added to the `item_lineage` table.
2. Submit COVID documents into your local system to populate the lineages.
3. Use the List of Deliveries Endpoint to get valid delivered reports.
4. Make a `POST` call to `waters/report/facilities`, with the body consisting of comma-separated report ids from the previous step.
5. Output should be a list of Facilities in the following format: 
```
{
    "meta": {
        "type": "DeliveryFacility",
        "totalCount": 1
    },
    "data": [
        {
            "dateSent": "2023-05-01T15:19:07.557Z",
            "orderingProvider": null,
            "performingFacility": "Any lab USA",
            "submitter": "hcintegrations",
            "reportId": "fd9721ce-cdd0-4d08-abd5-b8e989a9eda6"
        }
    ]
}
```
6. Using the following query parameters, repeat your queries and compare the outputs:
```
sortDir: (values: ASC or DESC)
sortCol: (values: DATE, PROVIDER, FACILITY, SUBMITTER, REPORT_ID)
since:   (dateTime in this format: 2023-04-12T19:00:00.307109Z)
until:   (dateTime in this format: 2024-04-12T19:00:00.307109Z)
```

## Changes
- Added new "origin" columns to the `item_lineage` table, plus new DB calls to use them.
- Added logic to pass the origin values into descendants as they are created.
- Added new endpoint to get bulk delivery facilities.

## Linked Issues
- #8985 
- #9265
- #9266
